File: Algorithms/Math/Prime factorization.py
import math


class PrimeFactorization(object):
    num = None
    prime_factorization_arr = []

    @classmethod
    def factorization(cls, data):
        if data > 2:
            cls.num = data
            cls.prime_factorization_arr = [True if i > 1 else False for i in range(data + 1)]
            cls.prime_numbers()
            if cls.prime_factorization_arr[data]:
                return f"{cls.num } is prime number!"
            string_return = f"the factorization is:\n {cls.num} = "
            for i in range(cls.num - 1):
                if cls.prime_factorization_arr[i] and data % i == 0:
                    pwer = 0
                    while data % i == 0:
                        pwer += 1
                        data = data / i
                    string_return += f"{i}^{pwer}"
                    if data < 2:
                        break
                    string_return += " * "
            return string_return

        else:
            return "The input you entered is incorrect!"

    @classmethod
    def prime_numbers(cls):
        sqrt_num = int(math.sqrt(cls.num))
        if math.sqrt(cls.num) > sqrt_num:
            sqrt_num += 1
        for prime in range(2, sqrt_num+1):
            if cls.prime_factorization_arr[prime]:
                for i in range(2*prime, cls.num+1, prime):
                    cls.prime_factorization_arr[i] = False


def main():  # test
    num = int(input("please enter number\nnum = "))
    print(PrimeFactorization.factorization(num))
    return 0


if __name__ == '__main__':
    main()

File: Algorithms/Math/Prime factorization.cpp
#include <iostream>
#include <math.h>

using namespace std;

int input_and_check(string print_text);
void prime_numbers(int prime_numbers_arr[], int num);
void prime_factorization(int prime_factorization_arr[], int num);

int main() {
	int num = 0;
	num = input_and_check("please enter number");
	int* prime_factorization_arr;
	prime_factorization_arr = (int*)malloc((num + 1) * sizeof(int));
	if (prime_factorization_arr == NULL)
		return 1;
	prime_factorization(prime_factorization_arr, num);
	cout << endl;
	return 1;
}

int input_and_check(string print_text) {
	int x = -1;
	cout << print_text << endl;
	cin >> x;
	while (!cin.good() || x < 2) {
		cout << "The input you entered is incorrect!" << endl << "Please try again" << endl;
		cin.clear();
		cin.ignore(INT_MAX, '\n');
		cin >> x;
	}
	return x;
}

void prime_numbers(int prime_numbers_arr[], int num) {
	int sqrt_num = (sqrt(num) + 0.5);
	for (int i = 0; i < num+1; ++i) prime_numbers_arr[i] = 1;
	prime_numbers_arr[0] = prime_numbers_arr[1] = 0;
	
	for (int prime = 2; prime <= sqrt_num; ++prime) {
		if (prime_numbers_arr[prime])
			for (int i = 2 * prime; i <= num; i += prime) prime_numbers_arr[i] = 0;
	}
}

void prime_factorization(int prime_factorization_arr[], int num) {
	int temp = num;
	prime_numbers(prime_factorization_arr, num);
	if (prime_factorization_arr[num]) {cout << num << " is prime number"; return;}
	cout << "the factorization is" << endl << num << " = ";
	for (int i = 0; i <= num; ++i)
		if (prime_factorization_arr[i] && temp % i == 0) {
			int pwer = 0;
			while (temp % i == 0) { ++pwer; temp = temp / i; }
			cout << i << '^' << pwer;
			if (temp < 2) break;
			cout << " * ";
		}

}